File: app/type/model/node.go
package model

import "time"

type Node struct {
	ID          uint64    `gorm:"primaryKey;AUTO_INCREMENT;column:id;" json:"id" example:"1"` // ID
	ProcessID   uint64    `gorm:"column:process_id;default:0" json:"process_id" example:"1"`
	ProcessCode string    `gorm:"column:process_code;default:''" json:"process_code" example:"1"`
	Tag         string    `gorm:"column:tag;default:''" json:"tag" example:"1"`
	Name        string    `gorm:"column:name;default:''" json:"name" example:"1"`
	Code        string    `gorm:"column:code;default:'';unique" json:"code" example:"1"`
	Type        string    `gorm:"column:type;default:'start'" json:"type" example:"start"`
	Description string    `gorm:"column:description;default:''" json:"description" example:"1"`
	CreatedBy   uint64    `gorm:"column:created_by;default:0" json:"created_by" example:"1"`
	UpdatedBy   uint64    `gorm:"column:updated_by;default:0" json:"updated_by" example:"1"`
	CreatedAt   time.Time `gorm:"column:created_at;autoCreateTime;" json:"created_at" example:"2020-10-11T10:10:10"` // 创建时间
	UpdatedAt   time.Time `gorm:"column:updated_at;autoUpdateTime;" json:"updated_at" example:"2020-10-11T10:10:10"` // 更新时间
}

func (*Node) TableName() string {
	return "node"
}

File: web/app/types/node.ts
import type { AssignType } from '~/types/assignment'

export type NodeType = 'start' | 'end' | 'user_task' | 'join'

export interface Node {
  id: number
  process_id: number
  process_code: string
  tag: string
  name: string
  code: string
  type: string
  description: string | null
  created_by: number
  created_by_name: string
  updated_by: number
  updated_by_name: string
  created_at: string
  updated_at: string
  x: number
  y: number
  assignType: AssignType
  assignTo: string[]
}

export interface NodeTypeConfig {
  name: string
  icon: string
  uiColor: 'primary' | 'secondary' | 'success' | 'info' | 'warning' | 'error' | 'neutral'
  canvasFill: string
}

export const nodeTypeConfigs: Record<NodeType, NodeTypeConfig> = {
  start: {
    name: '开始',
    icon: 'i-lucide-play',
    uiColor: 'success',
    canvasFill: 'fill-emerald-500'
  },
  end: {
    name: '结束',
    icon: 'i-lucide-flag',
    uiColor: 'error',
    canvasFill: 'fill-rose-500'
  },
  user_task: {
    name: '用户任务',
    icon: 'i-lucide-user-check',
    uiColor: 'primary',
    canvasFill: 'fill-sky-500'
  },
  join: {
    name: '汇聚',
    icon: 'i-lucide-git-merge',
    uiColor: 'warning',
    canvasFill: 'fill-amber-500'
  }
}

export interface NodeListBody {
  process_id?: number
  process_code?: string
  code: string
  type: string
}

export interface NodeAddBody {
  process_id: number
  process_code?: string
  tag: string
  name: string
  code: string
  type: string
  description: string
}

export interface NodeUpdateBody {
  id: number
  tag: string
  description: string
}
